Xamarin.Forms.Platform.iOS/Material/MaterialActivityIndicatorRenderer.cs

```diff
@@ -0,0 +1,129 @@
+using System.ComponentModel;
+using CoreGraphics;
+using MaterialComponents;
+using UIKit;
+using Xamarin.Forms;
+using MActivityIndicator = MaterialComponents.ActivityIndicator;
+
+[assembly: ExportRenderer(typeof(Xamarin.Forms.ActivityIndicator), typeof(Xamarin.Forms.Platform.iOS.Material.MaterialActivityIndicatorRenderer), new[] { typeof(VisualRendererMarker.Material) })]
+
+namespace Xamarin.Forms.Platform.iOS.Material
+{
+	public class MaterialActivityIndicatorRenderer : ViewRenderer<ActivityIndicator, MActivityIndicator>
+	{
+		SemanticColorScheme _defaultColorScheme;
+		SemanticColorScheme _colorScheme;
+
+		public MaterialActivityIndicatorRenderer()
+		{
+			VisualElement.VerifyVisualFlagEnabled();
+		}
+
+		protected override void OnElementChanged(ElementChangedEventArgs<ActivityIndicator> e)
+		{
+			_colorScheme?.Dispose();
+			_colorScheme = CreateColorScheme();
+
+			base.OnElementChanged(e);
+
+			if (e.NewElement != null)
+			{
+				if (Control == null)
+				{
+					_defaultColorScheme = CreateColorScheme();
+
+					SetNativeControl(CreateNativeControl());
+				}
+
+				UpdateColor();
+				UpdateIsRunning();
+
+				ApplyTheme();
+			}
+		}
+
+		protected virtual SemanticColorScheme CreateColorScheme()
+		{
+			return MaterialColors.Light.CreateColorScheme();
+		}
+
+		protected virtual void ApplyTheme()
+		{
+			ActivityIndicatorColorThemer.ApplySemanticColorScheme(_colorScheme, Control);
+		}
+
+		protected override MActivityIndicator CreateNativeControl()
+		{
+			return new MActivityIndicator
+			{
+				IndicatorMode = ActivityIndicatorMode.Indeterminate
+			};
+		}
+
+		protected override void OnElementPropertyChanged(object sender, PropertyChangedEventArgs e)
+		{
+			base.OnElementPropertyChanged(sender, e);
+
+			var updatedTheme = false;
+			if (e.PropertyName == ActivityIndicator.ColorProperty.PropertyName)
+			{
+				UpdateColor();
+				updatedTheme = true;
+			}
+			else if (e.PropertyName == ActivityIndicator.IsRunningProperty.PropertyName)
+			{
+				UpdateIsRunning();
+			}
+
+			if (updatedTheme)
+				ApplyTheme();
+		}
+
+		protected override void SetBackgroundColor(Color color)
+		{
+			// Do not call base to avoid the actual background view changing color.
+			//base.SetBackgroundColor(color);
+
+			if (Control == null)
+				return;
+
+			// TODO: Investigate whether or not we want to look for the track
```

Comment:
So wait, if they change the background color the track will just disappear?

Reply:
Yes and no. Right now, the default is no track. If you set **any** color, it will appear, but it will be the same color as the progress indicator (as it does not support custom colors). If you set it to transparent, it just disappears.